pkg/api/v1alpha1/tinkerbelltemplateconfig_defaults.go

```diff
@@ -29,39 +31,39 @@ warnings:
 `
 )
 
-func GetDefaultActionsFromBundle(b v1alpha1.VersionsBundle) []ActionOpt {
+func GetDefaultActionsFromBundle(b v1alpha1.VersionsBundle, disk string) []ActionOpt {
 	return []ActionOpt{
-		withStreamImageAction(b),
-		withNetplanAction(b),
-		withTinkCloudInitAction(b),
-		withDsCloudInitAction(b),
-		withKexecAction(b),
+		withStreamImageAction(b, disk),
+		withNetplanAction(b, disk),
+		withTinkCloudInitAction(b, disk),
+		withDsCloudInitAction(b, disk),
+		withKexecAction(b, disk),
 	}
 }
 
-func withStreamImageAction(b v1alpha1.VersionsBundle) ActionOpt {
+func withStreamImageAction(b v1alpha1.VersionsBundle, disk string) ActionOpt {
 	return func(a *[]tinkerbell.Action) {
 		*a = append(*a, tinkerbell.Action{
 			Name:    "stream-image",
 			Image:   b.Tinkerbell.TinkerbellStack.Actions.ImageToDisk.URI,
 			Timeout: 360,
 			Environment: map[string]string{
 				"IMG_URL":    b.EksD.Raw.Ubuntu.URI,
-				"DEST_DISK":  "/dev/sda",
+				"DEST_DISK":  disk,
 				"COMPRESSED": "true",
 			},
 		})
 	}
 }
 
-func withNetplanAction(b v1alpha1.VersionsBundle) ActionOpt {
+func withNetplanAction(b v1alpha1.VersionsBundle, disk string) ActionOpt {
 	return func(a *[]tinkerbell.Action) {
 		*a = append(*a, tinkerbell.Action{
 			Name:    "write-netplan",
 			Image:   b.Tinkerbell.TinkerbellStack.Actions.WriteFile.URI,
 			Timeout: 90,
 			Environment: map[string]string{
-				"DEST_DISK": "/dev/sda2",
+				"DEST_DISK": fmt.Sprintf("%s2", disk),
```

Comment:
We don't know this will be the second disk 100% of the time

Reply:
This is the 2nd disk partition not the 2nd disk. The ubuntu image we build has UEFI is the primary partition and the actual os on the 2nd partition. That's why we are appending 2 to the disk name